core/modules/qana/PostPlugin.cc

```diff
@@ -157,47 +157,49 @@ PostPlugin::applyPhysical(QueryPlugin::Plan& plan,
             auto const& expr = ordByTerm.getExpr();
             query::ValueExpr::FactorOpVector& factorOps = expr->getFactorOps();
             for (auto& factorOp:factorOps) {
-                auto factor = factorOp.factor;
-                query::ColumnRef::Ptr ordByColRef = factor->getColumnRef();
-                std::string orderByDb = ordByColRef->db;
-                std::string orderByTbl = ordByColRef->table;
-                std::string orderByCol = ordByColRef->column;
-                LOGS(_log, LOG_LVL_DEBUG, "Order By factorOp->factor=" << factor
-                     << " db=" << orderByDb << " tbl=" << orderByTbl << " col=" << orderByCol);
-
-                // If db or table is not empty, that's an error as those typically will not
-                // be passed back to the proxy in the table column name and the proxy will
-                // throw an error when it can't locate the ORDER BY argument in the result columns.
-                if (!orderByDb.empty() || !orderByTbl.empty()) {
-                    if (!orderByTbl.empty()) orderByTbl += ".";
-                    std::string msg = "ORDER BY argument should not include database or table for "
-                        + orderByDb + " " + orderByTbl + orderByCol + ".\n Remove qualifiers or "
-                        "use an alias.\n  ex:'SELECT a.val AS aVal ... ORDER BY aVal";
-                    LOGS(_log, LOG_LVL_WARN, msg);
-                    throw AnalysisError(msg);
-                }
+                query::ColumnRef::Vector orderByColumnRefVec;
+                factorOp.factor->findColumnRefs(orderByColumnRefVec);
```

Comment:
this is the key change; `getColumnRef` returns the ColumnRef value of the ValueFactor, *if* the ValueFactor is of type column ref. `findColumnRefs` recursively gets all the column refs used by the value factor (e.g. if the ValueFactor is of type Function and the func uses one or more ColumnRef arguments, it will return all the column ref arguments used by the function).